File: r-programming/rock-paper-scissors-function.R
cpu_choices <- c("rock", "paper", "scissors")

rockpaperscissor <- function(userchoice){
  
  if(!userchoice %in% cpu_choices) {
    stop("Invalid choice! Please choose 'rock', 'paper', or 'scissors'.")
  }
  
  cpu_choice <- sample(cpu_choices, 1)
  
  if(userchoice == "rock") {
    if(cpu_choice == "rock") {
      result <- "Draw"
    } else if(cpu_choice == "paper") {
      result <- "CPU wins"
    } else {
      result <- "You win"
    }
  } 
  else if(userchoice == "paper") {
    if(cpu_choice == "paper") {
      result <- "Draw"
    } else if(cpu_choice == "scissors") {
      result <- "CPU wins"
    } else {
      result <- "You win"
    }
  } 
  else if(userchoice == "scissors") {
    if(cpu_choice == "scissors") {
      result <- "Draw"
    } else if(cpu_choice == "rock") {
      result <- "CPU wins"
    } else {
      result <- "You win"
    }
  }
  
  print(paste("You choose", userchoice))
  print(paste("CPU chooses", cpu_choice))
  print(result)
  
  return(result)
}

rockpaperscissor("paper")

File: python/rock-paper-scissors-function.py
import random

cpu_choices = ["rock", "paper", "scissor"]

def rock_paper_scissors():
    cpu_choice = random.choice(cpu_choices)
    user_choice = input("Enter your choice: ")

    print("You choose:", user_choice)
    print("CPU chooses:", cpu_choice)

    if user_choice == cpu_choice:
        print("It's a tie!")
    elif (user_choice == "rock" and cpu_choice == "scissor") or \
         (user_choice == "paper" and cpu_choice == "rock") or \
         (user_choice == "scissor" and cpu_choice == "paper"):
        print("You win!")
    else:
        print("You lose!")

rock_paper_scissors()
